### PR TITLE
Fix an error being thrown while hovering over row/column headers.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
The problem in #6926 was indeed caused by [a problem recognized by @wojciechczerniak](https://github.com/handsontable/handsontable/issues/6926#issuecomment-652934597).
In the usecase he provided the error is being thrown when there's a cursor hovering over a header, while the header is being re-rendered. This triggers the `mouseover` event on a freshly-added child od the `TH` element, when the `parentElement` of the child is not yet set.

My fix consists of the following:

1. I added two helpers: 
   1.  `isDetached` (checks if the provided element has a parent) in the DOM helpers,
   2. `getTargetParent` (gets the `event.target` parent from the `composedPath` method, if the `parentElement` doesn't exist - it does not work for IE, though) in the event helpers.
2. In the `manualColumnResize` and `manualRowResize` plugins (where the error in question happened), I added some conditions, which should prevent the error from being thrown, even on IE. Some logic gets skipped here, but it doesn't seem to have any unwanted effects on functionality.

Additionally, I renamed one private method in the `manual*Resize` plugins, as it was misleading, and rewrote one method in `manualRowResize`, which was not working properly.

### How has this been tested?
- Added two test cases, one for each of the plugins.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #6926 

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: src/plugins/manualColumnResize/manualColumnResize.js [hunk 1]
```diff
@@ -337,14 +338,14 @@ class ManualColumnResize extends BasePlugin {
    * @returns {boolean}
    */
   checkIfColumnHeader(element) {
```
Comment (budnix) [inline src/plugins/manualColumnResize/manualColumnResize.js:340]: This shape of the code reminds me of our `closest` helper. Locally I've replaced the whole method body with `return closest(element, ['THEAD'], this.hot.rootElement) ? true : false;`, and after run test, nothing goes red. Do you think that that part of the code can be enhanced in that way?

File: src/plugins/manualColumnResize/manualColumnResize.js [hunk 2]
```diff
@@ -376,8 +377,11 @@ class ManualColumnResize extends BasePlugin {
    * @param {MouseEvent} event The mouse event.
    */
   onMouseOver(event) {
-    if (this.checkIfColumnHeader(event.target)) {
-      const th = this.getTHFromTargetElement(event.target);
+    // Workaround for #6926 - if the `event.target` is temporarily detached, we can go straight the the parent.
+    if (this.checkIfColumnHeader(isDetached(event.target) ? getTargetParent(event) : event.target)) {
+      // Because of a problem described in #6926, the `event.target` element is sometimes (temporarily) detached.
+      // This workaround prevents the error from #6926 happening.
+      const th = event.target.tagName === 'TH' ? event.target : this.getClosestTHParent(getTargetParent(event));
```
Comment (budnix) [inline src/plugins/manualColumnResize/manualColumnResize.js:384]:
```suggestion
    if (isDetached(event.target)) {
      return;
    }

    if (this.checkIfColumnHeader(event.target)) {
      const th = this.getClosestTHParent(event.target);
```
I don't see any advantages of using the `composedPath` (in the form of `getTargetParent` helper) method of the event object. IMO skipping the detached elements should be enough for this fix. I've compared how the two approaches behave and didn't see any differences. Moreover, the usage of the mentioned API would increase code complexity.

In both cases, the error disappears. Sometimes in both cases, the handler is not showing at all.

### Playing with PR changes:
for rows:
PR
for columns:
PR-columns

### Playing with applied skipped detached nodes:
for rows:
Local
for columns:
Local-columns

File: src/plugins/manualRowResize/manualRowResize.js [hunk 3]
```diff
@@ -292,14 +293,14 @@ class ManualRowResize extends BasePlugin {
    * @returns {boolean}
    */
   checkIfRowHeader(element) {
```
Comment (budnix) [inline src/plugins/manualRowResize/manualRowResize.js:295]: As in the previous comment maybe this part of code could be replace to `return closest(element, ['TH', 'TBODY'], this.hot.rootElement) ? true : false;` WDYT?

File: src/plugins/manualRowResize/manualRowResize.js [hunk 4]
```diff
@@ -349,8 +350,11 @@ class ManualRowResize extends BasePlugin {
    * @param {MouseEvent} event The mouse event.
    */
   onMouseOver(event) {
-    if (this.checkIfRowHeader(event.target)) {
-      const th = this.getTHFromTargetElement(event.target);
+    // Workaround for #6926 - if the `event.target` is temporarily detached, we can go straight the the parent.
+    if (this.checkIfRowHeader(isDetached(event.target) ? getTargetParent(event) : event.target)) {
+      // Because of a problem described in #6926, the `event.target` element is sometimes (temporarily) detached.
+      // This workaround prevents the error from #6926 happening.
+      const th = event.target.tagName === 'TH' ? event.target : this.getClosestTHParent(getTargetParent(event));
```
Comment (budnix) [inline src/plugins/manualRowResize/manualRowResize.js:357]:
```suggestion
    if (isDetached(event.target)) {
      return;
    }

    if (this.checkIfRowHeader(event.target)) {
      const th = this.getClosestTHParent(event.target);
```

**Review by budnix — CHANGES_REQUESTED** [review]
I've left some thoughts in the code comments.

File: src/plugins/manualColumnResize/manualColumnResize.js (hunk 2 above)
Comment (jansiegel) [reply src/plugins/manualColumnResize/manualColumnResize.js:384]: You might be right - this problem happens on `mouseover`, so after skipping one event on a detached element, the next one will probably be triggered on an attached element. I'll do some additional testing, but your suggestion looks promising 👍

**Review by budnix — APPROVED** [review]
👍

**Pull request merged**